backend/the_paginator/src/graphql.rs

```diff
@@ -0,0 +1,194 @@
+                    &before,
+                )
+                .map_err(CursorInputBuildError::UndecodableBefore)?
+                .0,
+            ),
+            None => None,
+        };
+        let first = match self.first {
+            Some(first) => Some(
+                first
+                    .try_into()
+                    .map_err(CursorInputBuildError::NegativeFirst)?,
+            ),
+            None => None,
+        };
+        let last = match self.last {
+            Some(last) => Some(
+                last.try_into()
+                    .map_err(CursorInputBuildError::NegativeLast)?,
+            ),
+            None => None,
+        };
+
+        Ok(QueryCursor::<Entity>::from_bounds(
+            after, before, first, last,
+        )?)
+    }
+}
+
+/// A [`Connection`] which produces pages of a Model
+#[allow(type_alias_bounds)]
+pub type ModelConnection<Model: ModelTrait> = Connection<
+    OpaqueCursor<<<Model::Entity as EntityTrait>::PrimaryKey as PrimaryKeyTrait>::ValueType>,
+    Model,
+    EmptyFields,
+    EmptyFields,
+    DefaultConnectionName,
+    DefaultEdgeName,
+    DisableNodesField,
+>;
+
+/// An error produces when primary key extraction fails
+#[derive(Debug, thiserror::Error)]
+#[error("Failed to extract the primary key")]
+pub struct PrimaryKeyExtractionError;
+
+fn try_extract_primary_key<Model>(
+    model: &Model,
+) -> Result<
+    <<Model::Entity as EntityTrait>::PrimaryKey as PrimaryKeyTrait>::ValueType,
+    PrimaryKeyExtractionError,
+>
+where
+    Model: ModelTrait,
+{
+    let columns = <Model::Entity as EntityTrait>::PrimaryKey::iter()
+        .map(|key| key.into_column())
+        .collect::<Vec<_>>();
+    let values = match columns.len() {
+        1 => Ok(ValueTuple::One(model.get(columns[0]))),
+        2 => Ok(ValueTuple::Two(
+            model.get(columns[0]),
+            model.get(columns[1]),
+        )),
+        3 => Ok(ValueTuple::Three(
+            model.get(columns[0]),
+            model.get(columns[1]),
+            model.get(columns[2]),
+        )),
+        4 => Ok(ValueTuple::Four(
+            model.get(columns[0]),
+            model.get(columns[1]),
+            model.get(columns[2]),
+            model.get(columns[3]),
+        )),
+        5 => Ok(ValueTuple::Five(
+            model.get(columns[0]),
+            model.get(columns[1]),
+            model.get(columns[2]),
+            model.get(columns[3]),
+            model.get(columns[4]),
+        )),
+        6 => Ok(ValueTuple::Six(
+            model.get(columns[0]),
+            model.get(columns[1]),
+            model.get(columns[2]),
+            model.get(columns[3]),
+            model.get(columns[4]),
+            model.get(columns[5]),
+        )),
+        _ => Err(PrimaryKeyExtractionError),
+    }?;
+    Ok(<<Model::Entity as EntityTrait>::PrimaryKey as PrimaryKeyTrait>::ValueType::from_value_tuple(
+            values,
+        ))
+}
```

Comment:
Would like to come up with a better alternative to this, but haven't been able to figure one out... The typical approach is a trait with (templated) implementations for each tuple width, but as far as I know there is no way to select on this when looking at an associated type. Any ideas?